src/app/lib/web_channel.js

```diff
@@ -0,0 +1,39 @@
+import Events from 'ampersand-events';
+
+class WebChannel {
+  constructor() {
+    // turn this into an event emitter
+    Events.createEmitter(this);
+
+    // listen for WebChannel messages
+    window.addEventListener('WebChannelMessageToContent', this._messageReceived.bind(this));
```

Comment:
Bit of context: this code was taken from https://github.com/mozilla/universal-search-addon, where we used `WebChannelMessageToContent` because it's the conventional event name used [by WebChannel.jsm](https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules/WebChannel.jsm#Chrome_code), and we use `WebChannel.jsm` in that project.

But yeah, let's definitely talk about CustomEvent names when we sort out the overall API :-)